File: app/src/main/java/com/kamelong/aodia/EditTrainType/EditTrainTypeView.java
package com.kamelong.aodia.EditTrainType;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.RadioGroup;

import com.kamelong.ColorPicker.ColorPickerDialog;
import com.kamelong.ColorPicker.ColorPickerListener;
import com.kamelong.OuDia.TrainType;
import com.kamelong.aodia.R;
import com.kamelong.tool.Color;
import com.kamelong.tool.SDlog;

public class EditTrainTypeView extends LinearLayout {
    public boolean checked = false;

    public EditTrainTypeView(final Context context, final TrainType trainType) {
        super(context);
        try {
            LayoutInflater.from(context).inflate(R.layout.traintype_edit_view, this);
            final CheckBox checkBox = findViewById(R.id.checkBox);
            checkBox.setOnCheckedChangeListener((buttonView, isChecked) -> checked = isChecked);
            final EditText name = findViewById(R.id.nameEdit);
            name.setText(trainType.name);
            name.setOnFocusChangeListener((view, b) -> trainType.name = name.getEditableText().toString());
            final EditText shortName = findViewById(R.id.shortNameEdit);
            shortName.setText(trainType.shortName);
            shortName.setOnFocusChangeListener((view, b) -> trainType.shortName = shortName.getEditableText().toString());

            final Button textColor = findViewById(R.id.textColor);
            textColor.setBackgroundColor(trainType.textColor.getAndroidColor());
            textColor.setOnClickListener(view -> {
                try {
                    new ColorPickerDialog(context, color -> {
                        textColor.setBackgroundColor(color.getAndroidColor());
                        trainType.textColor = color;
                    },trainType.textColor).show();
                } catch (Exception e) {
                    SDlog.log(e);
                    SDlog.toast("色選択時にエラーが発生しました");

                }
            });
            final Button diaColor = findViewById(R.id.diaColor);
            diaColor.setBackgroundColor(trainType.diaColor.getAndroidColor());
            diaColor.setOnClickListener(view -> {
                try {

                    new ColorPickerDialog(context, color -> {
                        diaColor.setBackgroundColor(color.getAndroidColor());
                        trainType.diaColor = color;
                    },trainType.diaColor).show();


                } catch (Exception e) {
                    SDlog.log(e);
                    SDlog.toast("色選択時にエラーが発生しました");
                }
            });
            RadioGroup diaStyle = findViewById(R.id.diaStyleRadio);
            switch (trainType.lineStyle) {
                case 0:
                    diaStyle.check(R.id.radioButton);
                    break;
                case 1:
                    diaStyle.check(R.id.radioButton2);
                    break;
                case 2:
                    diaStyle.check(R.id.radioButton4);
                    break;
                case 3:
                    diaStyle.check(R.id.radioButton3);
                    break;
            }
            diaStyle.setOnCheckedChangeListener((radioGroup, i) -> {
                switch (i) {
                    case R.id.radioButton:
                        trainType.lineStyle = 0;
                        break;
                    case R.id.radioButton2:
                        trainType.lineStyle = 1;
                        break;
                    case R.id.radioButton4:
                        trainType.lineStyle = 2;
                        break;
                    case R.id.radioButton3:
                        trainType.lineStyle = 3;
                        break;

                }
            });
            ((CheckBox) findViewById(R.id.checkBox2)).setChecked(trainType.bold);
            ((CheckBox) findViewById(R.id.checkBox2)).setOnCheckedChangeListener((compoundButton, b) -> trainType.bold = b);
            ((CheckBox) findViewById(R.id.checkBox3)).setChecked(trainType.stopmark);
            ((CheckBox) findViewById(R.id.checkBox3)).setOnCheckedChangeListener((compoundButton, b) -> trainType.stopmark = b);


        } catch (Exception e) {
            SDlog.log(e);
        }
    }
}

File: app/src/main/java/com/kamelong/aodia/editTrainType/EditTrainTypeView.kt
package com.kamelong.aodia.editTrainType

import android.content.Context
import android.text.Editable
import android.text.TextWatcher
import android.view.LayoutInflater
import android.view.View
import android.widget.*
import com.kamelong.aodia.R
import com.kamelong.aodia.diadata.AOdiaTrainType
import com.azeesoft.lib.colorpicker.ColorPickerDialog
import com.kamelong.tool.Color


class EditTrainTypeView(context: Context,trainType:AOdiaTrainType): FrameLayout(context) {
    val layout = LayoutInflater.from(context).inflate(R.layout.edit_traintype_view, this)

    init {
        layout.findViewById<EditText>(R.id.nameEdit).setText(trainType.name)
        layout.findViewById<EditText>(R.id.nameEdit).addTextChangedListener(object :TextWatcher{
            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
            }

            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
            }

            override fun afterTextChanged(edit: Editable?) {
                if(edit.toString().isNotEmpty()){
                    trainType.name=edit.toString()
                }
            }

        })
        layout.findViewById<EditText>(R.id.shortNameEdit).setText(trainType.shortName)
        layout.findViewById<EditText>(R.id.shortNameEdit).addTextChangedListener(object :TextWatcher{
            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
            }

            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
            }

            override fun afterTextChanged(edit: Editable?) {
                if(edit.toString().isNotEmpty()){
                    trainType.shortName=edit.toString()
                }
            }

        })
        layout.findViewById<Button>(R.id.textColor).setBackgroundColor(trainType.textColor.androidColor)
        layout.findViewById<Button>(R.id.textColor).setOnClickListener {
            val colorPickerDialog = ColorPickerDialog.createColorPickerDialog(context)
            colorPickerDialog.setLastColor(trainType.textColor.androidColor)
            colorPickerDialog.setOnColorPickedListener { color, hexVal ->
                trainType.textColor=Color(color)
                layout.findViewById<Button>(R.id.textColor).setBackgroundColor(trainType.textColor.androidColor)
            }
            colorPickerDialog.show()
        }
        layout.findViewById<Button>(R.id.diaColor).setBackgroundColor(trainType.diaColor.androidColor)
        layout.findViewById<Button>(R.id.diaColor).setOnClickListener {
            val colorPickerDialog = ColorPickerDialog.createColorPickerDialog(context)
            colorPickerDialog.setLastColor(trainType.diaColor.androidColor)
            colorPickerDialog.setOnColorPickedListener { color, hexVal ->
                trainType.diaColor= Color(color)
                layout.findViewById<Button>(R.id.diaColor).setBackgroundColor(trainType.diaColor.androidColor)
            }
            colorPickerDialog.show()
        }
        layout.findViewById<RadioGroup>(R.id.diaStyleRadio).check(
                when(trainType.lineStyle){
                    0->R.id.radioButton
                    1->R.id.radioButton2
                    2->R.id.radioButton4
                    3->R.id.radioButton3
                    else->R.id.radioButton
                }
        )
        layout.findViewById<RadioGroup>(R.id.diaStyleRadio).setOnCheckedChangeListener { radioGroup, index ->
            trainType.lineStyle=when(index){
                0->0
                1->1
                2->2
                3->3
                else->0
            }
        }
        layout.findViewById<CheckBox>(R.id.checkBox2).isChecked=trainType.lineBold
        layout.findViewById<CheckBox>(R.id.checkBox2).setOnCheckedChangeListener { compoundButton, boolean -> trainType.lineBold=boolean }
        layout.findViewById<CheckBox>(R.id.checkBox3).isChecked=trainType.showStop
        layout.findViewById<CheckBox>(R.id.checkBox3).setOnCheckedChangeListener { compoundButton, boolean -> trainType.showStop=boolean }







    }
}